File: FCISUI/Models/SvgMap.cs
namespace FCISUI.Models
{
    public partial class SvgMap
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string? Defs { get; set; }
        public string? BackgroundSvg { get; set; }
        public byte[]? BackgroundImage {get; set;}
        public string Viewbox { get; set; }

        public int FacilityId { get; set; }

        public virtual ICollection<SvgMapPin> SvgMapPins {get; set;}
        public virtual ICollection<SvgMapArrow> SvgMapArrows {get; set;}
    }
}

File: FCISUI/ClientApp/src/app/api/models/svgMap.ts
export interface SvgMapPin {
  id: number;
  locationId: string;
  title: string;
  cx: number;
  cy: number;
  r: number;
  path: string;
}

export interface SvgMapArrow {
  id: number;
  locationId: string;

  arrowOutlinePoints: string;
  arrowInsidePoints: string;
}

export interface SvgMap {
  id: number;
  name: string;
  defs: string;
  backgroundSvg: string;
  viewbox: string;
  facilityId: number;
  svgMapPins: SvgMapPin[];
  svgMapArrows: SvgMapArrow[];
}

export const defaultSvgMap: SvgMap = {
  id: 0,
  backgroundSvg: '',
  name: '',
  svgMapPins: [],
  svgMapArrows: [],
  viewbox: '0 0 0 0',
  defs: '',
  facilityId: 0
};
